spatialos/UnrealGDK
# Feature/unr 3315 connect local server to cloud

Adding a new **Connect local server worker to the cloud deployment** checkbox, in Editor Setting under Cloud Connection:
![image](https://user-images.githubusercontent.com/56884490/85114237-9fe49500-b210-11ea-8528-3d2d3c59d44c.png)


In addition, added the option to select the port and the listening address of the server receptionist proxy if needed.

If  **Connect local server worker to the cloud deployment** is enabled when pressing play to connect to a Cloud Deployment both the Client and Server will start and connect to the Deployment.

If **Connect local server worker to the cloud deployment** is disabled, only the Client will start and connect to the Deployment.

The server is connecting via a receptionist proxy server that starts when you click the Play button.

No automated tests are included. Tested locally by running both local and cloud deployments.

To test : 
- Click Cloud in the SpatialOS Toolbar
- Select the default launch configuration file : **"manual_worker_connection_only"** option set to **true.**
- Start the cloud deployment
- Navigate to SpatialOS Settings> **Connect local server worker to the cloud deployment**  enable it
- Press play.
- You should see that both a UnrealClient and a UnrealWorker connected to the cloud deployment.

Good to test all different workflows,  enable /disable Connect local server worker to the cloud, local deployment, cloud deployment as we run a proxy that can block the local deployment too.

#### Documentation
Release Note, [Feature Documentation ](https://docs.google.com/document/d/1uxFe4JkISvoWZsLJPSsCjvDtF_S4yKxA1esQAt2vkM0/edit?usp=sharing)
